src/parenthetics.py

```diff
@@ -0,0 +1,168 @@
+        """Add a tail to the end of the list."""
+        if self.tail is None:
+            self.push(val)
+        else:
+            new_node = DLLNODE(val)
+            self.tail.next = new_node
+            self.tail = new_node
+            self.size_of_list += 1
+
+    def shift(self):
+        """Remove node from end of list and return to user."""
+        if self.tail is None:
+            raise IndexError("Impossible to shift from an empty list.")
+        value_shifted = self.tail.val
+        self.tail = self.tail.prev
+        if hasattr(self.tail, "next"):
+            self.tail.next = None
+        self.size_of_list -= 1
+        return value_shifted
+
+    def search(self, val):
+        """Return the val of the node when searched."""
+        current = self.head
+        if current is None:
+            raise NameError("Parameter given, {}, is not in list".format(str(val)))
+        while current:
+            if current.val == val:
+                return current
+            current = current.next
+        return current
+
+    def remove(self, node_to_delete):
+        """Remove the node passed into the parameter."""
+        current_node = self.search(node_to_delete)
+        if current_node is self.head:
+            self.pop()
+        elif current_node is self.tail:
+            self.shift()
+        elif current_node and current_node.val == node_to_delete:
+            current_node.prev.next = current_node.next
+            current_node.next.prev = current_node.prev
+            self.size_of_list -= 1
+        else:
+            raise NameError("Parameter given, {}, is not in list".format(node_to_delete))
+
+
+class Queue(object):
+    """Define the Queue class.
+
+    It is important to note that the "tail" of our queue is actually the front of the Queue.
+
+    __init__(): initializes the Queue class as a composite of the DLL class.
+    enqueue(): uses the push method from DLL to add nodes to the end of the Queue.
+    dequeue(): removes a node from the front of the dequeue
+    peek(): Looks at the front value of the Queue without removing it or advancing the view beyond it.
+    size(): returns the size of the Queue.
+    """
+
+    def __init__(self, iterable=None):
+        """Initiate the Queue as an instance of DLL."""
+        self._container = DLL(iterable)
+
+    def enqueue(self, val):
+        """Add value to the Queue at the end."""
+        self._container.push(val)
+
+    def dequeue(self):
+        """Remove the first value of the queue."""
+        return self._container.shift()
+
+    def peek(self):
+        """Look at next value in the Queue without removing."""
+        return self._container.tail.val
+
+    def size(self):
+        """How big the Queue is."""
+        return self._container.size_of_list
+
+
+def proper_parenthetics(paren_string):
+    """The module adds a string, and evals whether the parens line up."""
+    new_q = Queue(paren_string)
+    if paren_string == "":
+        return 0
```

Comment:
This is a nice easy out, but you shouldn't need to do this checking. It should be easily handled by whatever loop you've established below.